nncf/quantization/algorithms/channel_alignment/backend.py

```diff
@@ -0,0 +1,144 @@
+        :return: Bias value in the NumPy format.
+        """
+
+    @staticmethod
+    @abstractmethod
+    def get_weight_value(node: NNCFNode, model: TModel, port_id: int) -> np.ndarray:
+        """
+        Returns bias value in the NumPy format of provided node.
+
+        :param node: Node of NNCFGraph with bias value.
+        :param model: Backend-specific model for the initializer finding.
+        :param nncf_graph: NNCFGraph instance with the node.
+        :return: Bias value in the NumPy format.
+        """
+
+    @staticmethod
+    @abstractmethod
+    def get_activation_port_ids_for_node(node: NNCFNode) -> Tuple[int, int]:
+        """
+        Returns Input Port ID and Output Port ID corresponding to activation input and output edges for
+        the node.
+        Supports only nodes that could have bias value.
+
+        :param node: Node of NNCFGraph with bias value.
+        """
+
+    @staticmethod
+    @abstractmethod
+    def get_weights_port_ids_for_node(node: NNCFNode) -> Tuple[int, int]:
+        """
+        Returns Input Port ID and Output Port ID corresponding to node weights inputr port id and
+        constant output port id the node.
+
+        :param node: Node of NNCFGraph.
+        """
+
+    @staticmethod
+    @abstractmethod
+    def get_statistic_collector(
+        reduction_shape, q: float, num_samples: int, inplace: bool
+    ) -> TensorStatisticCollectorBase:
+        """
+        Get backend-specific tensor collector that collects medians of minimal and maximal quantiles.
+
+        :param reduction_shape: Target reduction shape for the reduction.
+        :param q: Minimal quantile for the tensor collector.
+        :param num_samples: Num samples to collect by the tensor collector.
+        :param inplace: Should statistic be calculated inplace or out of place.
+        :return: Backend-specific tensor collector that collects medians of minimal and maximal quantiles.
+        """
+
+    @staticmethod
+    @abstractmethod
+    def is_node_with_bias(node: NNCFNode, nncf_graph: NNCFGraph) -> bool:
+        """
+        Checks if the node has a bias or not.
+
+        :param node: The node to check.
+        :param nncf_graph: The NNCF graph.
+        :return: True` if `node` corresponds to the operation with bias
+            (bias is added to the output tensor of that operation), `False` otherwise.
+        """
+
+    @staticmethod
+    @abstractmethod
+    def get_dims_descriptor(node: NNCFNode) -> LayoutDescriptor:
+        """
+        Return weights layout descriptor of the given node if it is possible and None otherwise.
+        Only convolutional and linear nodes are supported.
+
+        :param node: NNCFNode to get layout descriptor from.
+        :return: Weights layout descriptor of the given node if it is possible and None otherwise.
+        """
+
+    @staticmethod
+    @abstractmethod
+    def get_conv_layer_attributes(node: NNCFNode) -> Optional[ConvolutionLayerAttributes]:
```

Comment:
This is very ugly method, you need to think how to move it to the common part of algorithm.